src/main/java/com/stripe/model/Card.java

```diff
@@ -1,107 +1,365 @@
+  }
 
-  // <editor-fold desc="delete">
   /**
-   * Delete a card.
+   * Delete a specified external account for a given account.
+   *
+   * <p>Delete a specified source for a given customer.
    */
-  @Override
   public Card delete() throws StripeException {
-    return delete((RequestOptions) null);
+    return delete((Map<String, Object>) null, (RequestOptions) null);
   }
 
   /**
-   * Delete a card.
+   * Delete a specified external account for a given account.
+   *
+   * <p>Delete a specified source for a given customer.
    */
-  @Override
   public Card delete(RequestOptions options) throws StripeException {
-    return request(RequestMethod.DELETE, this.getInstanceUrl(), null, Card.class, options);
+    return delete((Map<String, Object>) null, options);
   }
-  // </editor-fold>
 
-  // <editor-fold desc="update">
   /**
-   * Update a card.
+   * Delete a specified external account for a given account.
+   *
+   * <p>Delete a specified source for a given customer.
    */
-  @Override
-  public Card update(Map<String, Object> params) throws StripeException {
-    return update(params, (RequestOptions) null);
+  public Card delete(Map<String, Object> params) throws StripeException {
+    return delete(params, (RequestOptions) null);
   }
 
   /**
-   * Update a card.
+   * Delete a specified external account for a given account.
+   *
+   * <p>Delete a specified source for a given customer.
    */
-  @Override
-  public Card update(Map<String, Object> params, RequestOptions options) throws StripeException {
-    return request(RequestMethod.POST, this.getInstanceUrl(), params, Card.class, options);
-  }
-  // </editor-fold>
-
-  @Override
-  protected String getInstanceUrl() {
-    String result = super.getInstanceUrl();
-    if (result != null) {
-      return result;
-    } else if (this.getRecipient() != null) {
-      return String.format("%s/%s/cards/%s", classUrl(Recipient.class), this.getRecipient(),
-          this.getId());
+  public Card delete(Map<String, Object> params, RequestOptions options) throws StripeException {
+    String url;
+    if (this.getAccount() != null) {
+      url =
+          String.format(
+              "%s%s",
+              Stripe.getApiBase(),
+              String.format(
+                  "/v1/accounts/%s/external_accounts/%s", this.getAccount(), this.getId()));
+    } else if (this.getCustomer() != null) {
+      url =
+          String.format(
+              "%s%s",
+              Stripe.getApiBase(),
+              String.format("/v1/customers/%s/sources/%s", this.getCustomer(), this.getId()));
     } else {
-      return null;
+      throw new InvalidRequestException(
+          "Unable to construct url because [account, customer] field(s) are all null",
+          null,
+          null,
+          null,
+          0,
+          null);
     }
+    return request(ApiResource.RequestMethod.DELETE, url, params, Card.class, options);
```

Comment:
16) Delete (standard): Instance method to delete a sub-resource on itself, relying on ID attributes of the instance, and of its parent resource. There is validation on parent resource ID.

Reply:
https://github.com/stripe/stripe-java/blob/2aa5db5c39dc6e91228507798083f373904cc8c0/src/main/java/com/stripe/model/Card.java#L334-L364